facebook/react-native
# [FlatList] Add regression test for context api

See #18351 for details about the stack overflow when a FlatList is used within a context. That issue should be fixed by the commit 4f8328bf2fb71b104baa70178a1e1f9636773c6a, and this PR will ensure future regressions here will be caught.

# Test Plan

1. Check out `facebook/react-native` commit 321b107685e8f5d3bd36f673d81041ec3876b34e (or any recent commit prior to 4f8328bf2fb71b104baa70178a1e1f9636773c6a)
2. Run `RNTester` on iOS.
3. Select the `FlatListExample`.
4. Confirm that the failure highlighted in #18351 is reproduced:

![simulator screen shot - iphone x - 2018-03-14 at 11 44 10](https://user-images.githubusercontent.com/165856/37425171-ac49bdb6-277f-11e8-875f-88a0529f06c1.png)


5. Rebase on top of master (ensuring 4f8328bf2fb71b104baa70178a1e1f9636773c6a is applied)
6. Refresh the iOS Simulator to load the changes (CMD + R)
7. Confirm the Flat List Example is rendered and no Redbox is shown.

![simulator screen shot - iphone x - 2018-03-14 at 11 58 09](https://user-images.githubusercontent.com/165856/37425169-ac24a6de-277f-11e8-925b-257b20947381.png)
